pkg/controller/controllerutil/controllerutil.go

```diff
@@ -89,10 +89,10 @@ func SetControllerReference(owner, controlled metav1.Object, scheme *runtime.Sch
 	return nil
 }
 
-// EnsureOwnerReference is a helper method to make sure the given object contains
+// SetOwnerReference is a helper method to make sure the given object contains
 // an object reference to the object provided.
-// If a reference already exists, it'll be overwritten with the newly provided version.
-func EnsureOwnerReference(owner, object metav1.Object, scheme *runtime.Scheme) error {
+// If a reference to the same object already exists, it'll be overwritten with the newly provided version.
```

Comment:
With this change, maybe it would be good to explain in the comments why you would use `SetOwnerReference` over `SetControllerReference`.

Reply:
What would you like that to look like?

Reply:
:scream: words escape me. Here is my best attempt:

> This allows you to declare that owner has a dependency on the object without specifying it as a controller.